cmd/minikube/cmd/start.go

```diff
@@ -141,6 +145,15 @@ func runStart(cmd *cobra.Command, args []string) {
 		glog.Errorln("Error starting the VM: ", err)
 		os.Exit(1)
 	}
+
+	// Set Proxy to as Shell Env
+	if viper.IsSet("http-proxy") || viper.IsSet("https-proxy") {
```

Comment:
@hferentschik  Why I am actually not in favor to move registration bits and this one to StartHost because https://github.com/minishift/minishift/blob/master/cmd/minikube/cmd/start.go#L139 so whatever we move cluster.StartHost it will going to provision 3 time in case of any error. Since registration take around 1-2 mins if that fail first time the it start retry (assume user provided invalid username/password). 

MachineConfig already have Docker Env which actually responsible ( https://github.com/minishift/minishift/blob/master/pkg/minikube/cluster/cluster.go#L172-L180 ) for set across different provisioner => https://github.com/minishift/minishift/blob/master/cmd/minikube/cmd/start.go#L139